src/System.Threading.Thread/tests/ThreadTests.netstandard.cs

```diff
@@ -20,10 +20,21 @@ public static class ThreadTests
         [Fact]
         public static void ConstructorTest()
         {
-            var t = new Thread(() => { });
-            t = new Thread(() => { }, 0);
-            t = new Thread(state => { });
-            t = new Thread(state => { }, 0);
+            Action<Thread> startThreadAndJoin =
+                t =>
+                {
+                    t.IsBackground = true;
+                    t.Start();
+                    Assert.True(t.Join(UnexpectedTimeoutMilliseconds));
+                };
+            startThreadAndJoin(new Thread(() => { }));
+            startThreadAndJoin(new Thread(() => { }, 0));
+            startThreadAndJoin(new Thread(() => { }, 64 << 10)); // 64 KB
+            startThreadAndJoin(new Thread(() => { }, 16 << 20)); // 16 MB
```

Comment:
Would it make sense to verify that the stack is really 16MB by doing stackalloc for close to 16MB of stuff?